### DISCUSSION TITLE
Different results within the self-hosted version

### DISCUSSION THREAD

**Discussion by sathishlakshmanan** [osm-search/Nominatim#3354] [Q&A] [answered]
I have self-hosted Nominatim geocoding server on Ubuntu 22.04 following this post `https://www.linuxbabe.com/ubuntu/osm-nominatim-geocoding-server-ubuntu-22-04` and also setup a web server with Apache.

As you can see in the results below, fields like `name`, `addresstype` are present when queried via CLI but missing when queried via Apache web server. I would like to know why this is happening and how can this be rectified. TIA

```
nominatim search --query berlin --limit 1 --format json

[
    {
      "place_id": 131799146,
      "licence": "Data © OpenStreetMap contributors, ODbL 1.0. http://osm.org/copyright",
      "osm_type": "relation",
      "osm_id": 62422,
      "lat": "52.5170365",
      "lon": "13.3888599",
      "class": "boundary",
      "type": "administrative",
      "place_rank": 8,
      "importance": 0.7875390282491362,
      "addresstype": "city",
      "name": "Berlin",
      "display_name": "Berlin, Deutschland",
      "boundingbox": [
          "52.3382448",
          "52.6755087",
          "13.0883450",
          "13.7611609"
      ]
  }
]
```

```
curl 'http://localhost:8080/nominatim/search?q=berlin&limit=1&format=json'

[
  {
    "place_id": 131799146,
    "licence": "Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright",
    "osm_type": "relation",
    "osm_id": 62422,
    "boundingbox": [
      "52.3382448",
      "52.6755087",
      "13.088345",
      "13.7611609"
    ],
    "lat": "52.5170365",
    "lon": "13.3888599",
    "display_name": "Berlin, Deutschland",
    "class": "boundary",
    "type": "administrative",
    "importance": 0.8875390282491362
  }
]
```
Status:
```
{
  "status": 0,
  "message": "OK",
  "data_updated": "2024-02-19T00:56:55+00:00",
  "software_version": "4.3.2-0",
  "database_version": "4.3.2-0"
}
```

FYI, I have also referred to this discussion `https://github.com/osm-search/Nominatim/discussions/3287` and other similar discussions but nothing is related to this issue.

**Comment by mtmail** [comment, marked as answer]
Nominatim is being rewritten from PHP to Python. Huge project and can lead to confusion. In Nominatim 4.3 the CLI runs against the Python code, the website by default (and based on the tutorial you linked) against PHP. See https://nominatim.org/release-docs/latest/admin/Deployment-Python/ how to switch to the Python frontend.

Current plan is that Nominatim 4.4 will make the Python frontend default. (https://nominatim.org/2023/12/19/roadmap-nominatim5.html)
